ros_packages/voice_assistant/voice_assistant/chat.py

```diff
@@ -81,51 +83,104 @@ def chat(self, goal_handle: ServerGoalHandle):
-                    curr_sentence = ""
-                curr_sentence += token
+                    prev_text = None
+                    prev_text_type = None
+
+                # add token to current text; remove leading white-spaces, if current-text is empty
+                curr_text = curr_text + (token if len(curr_text) > 0 else token.lstrip())
+
+                while True: # loop until current-text was not stripped during current iteration
+
+                    # if the goal was cancelled, return immediately
+                    if goal_handle.is_cancel_requested:
+                        goal_handle.canceled()
+                        return Chat.Result() # TODO
+                    
+                    # check if the collected text is visual-code
+                    code_visual_match = code_visual_pattern.search(curr_text)
+                    if code_visual_match is not None:
+                        # extract the visual-code by removing the opening + closing tag and store it as previous text
+                        code_visual = code_visual_match.group(1)
+                        prev_text = code_visual
+                        prev_text_type = Chat.Goal.TEXT_TYPE_CODE_VISUAL
+                        # create a chat message from the visual-code, including opening and closing tags
+                        chat_message_text = code_visual_match.group(0)
+                        self.executor.create_task(self.create_chat_message, chat_id, chat_message_text, False)
+                        # strip the current text
+                        curr_text = curr_text[code_visual_match.end():].rstrip()
+                        continue
+
+                    # check if collected text is a sentence
+                    sentence_match = sentence_pattern.search(curr_text)
+                    if sentence_match is not None:
+                        # extract the visual-code by removing the opening + closing tag and store it as previous text
+                        sentence = sentence_match.group(1) + (sentence_match.group(3) if sentence_match.group(3) is not None else "")
+                        prev_text = sentence
+                        prev_text_type = Chat.Goal.TEXT_TYPE_SENTENCE
+                        # create a chat message from the visual-code, including opening and closing tags
+                        chat_message_text = sentence
+                        self.executor.create_task(self.create_chat_message, chat_id, chat_message_text, False)
+                        # strip the current text
+                        curr_text = curr_text[sentence_match.end(3 if sentence_match.group(3) is not None else 1):].rstrip()
+                        continue
+                        
+                    break
 
             # create chat-message for remaining input
-            self.executor.create_task(self.create_chat_message, chat_id, curr_sentence, False)
+            if (len(curr_text) > 0):
```

Comment:
no parantheses for `if` statements in python

Reply:
removed parantheses